builder/builder.go

```diff
@@ -26,24 +27,94 @@ type Builder struct {
 	bc         *blockchain.Blockchain
 	vm         vm.VM
 	newHeads   *feed.Feed[*core.Header]
-	log        utils.Logger
+	log        utils.SimpleLogger
+
+	PendingBlock *blockchain.Pending
 }
 
-func New(ownAddr *felt.Felt, bc *blockchain.Blockchain, builderVM vm.VM, log utils.Logger) *Builder {
+func New(ownAddr *felt.Felt, bc *blockchain.Blockchain, builderVM vm.VM, log utils.SimpleLogger) *Builder {
 	return &Builder{
 		ownAddress: *ownAddr,
 
 		bc:       bc,
 		vm:       builderVM,
 		newHeads: feed.New[*core.Header](),
+		log:      log,
 	}
 }
 
 func (b *Builder) Run(ctx context.Context) error {
-	<-ctx.Done()
+	if err := b.InitPendingBlock(); err != nil {
+		return err
+	}
+
+	blockInterval := 5 * time.Second
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-time.After(blockInterval):
+			if err := b.Finalise(); err != nil {
+				return err
+			}
+		}
+	}
+}
+
+// InitPendingBlock initialises a new pending block
+func (b *Builder) InitPendingBlock() error {
```

Comment:
I think we can un-export this.

Reply:
It kinda helps with testing. Allows us to not deal with services in tests.